src/components/API.js

```diff
@@ -0,0 +1,108 @@
+class Api {
+  constructor(options) {
+    this._options = options;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._options.baseUrl}/cards`, {
+      headers: this._options.headers,
+    }).then(this._handleResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._options.baseUrl}/users/me`, {
+      headers: this._options.headers,
+    }).then(this._handleResponse);
+  }
+
+  updateUserProfile(userData) {
+    return fetch(`${this._options.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._options.headers,
+      body: JSON.stringify(userData),
+    }).then(this._handleResponse);
+  }
+  _handleResponse(res) {
+    if (!res.ok) {
+      return Promise.reject(`Error, ${res.status}`);
+    }
+    return res.json();
+  }
+
+  editProfileImage(link) {
+    // receive link
+    return fetch(`${this._options.baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._options.headers,
+      body: JSON.stringify({ avatar: link }),
+    })
+      .then(this._handleResponse)
+      .catch((err) => {
+        console.error(err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.